src/Client.ts

```diff
@@ -220,24 +206,22 @@ export default class Client {
   ): Promise<PublicKeyBundle | undefined> {
     // have to avoid undefined to not trip TS's strictNullChecks on recipientKey
     let recipientKey: PublicKeyBundle | null = null
-    await this.waku.store.queryHistory([buildUserContactTopic(peerAddress)], {
-      pageSize: 5,
-      pageDirection: PageDirection.BACKWARD,
-      callback: (msgs: WakuMessage[]) => {
-        for (const msg of msgs) {
-          if (!msg.payload) continue
-          const bundle = ContactBundle.fromBytes(msg.payload as Uint8Array)
-          const keyBundle = bundle.keyBundle
-
-          const address = keyBundle?.walletSignatureAddress()
-          if (address === peerAddress) {
-            recipientKey = keyBundle
-            break
-          }
-        }
-        return recipientKey !== null
-      },
-    })
+    const stream = this.apiClient.queryIterator(
+      { contentTopics: [buildUserContactTopic(peerAddress)] },
+      { pageSize: 5 }
+    )
+
+    for await (const env of stream) {
+      if (!env.message) continue
+      const bundle = ContactBundle.fromBytes(b64Decode(env.message.toString()))
```

Comment:
I don't understand where does the `b64Decode` come from, are we changing the format of the contact bundles?

Reply:
Oh, is that the JSON encoding between the grpc gateway and the client?